src/main/java/calculator/CalculatorService.java

```diff
@@ -0,0 +1,132 @@
+package calculator;
+
+import com.google.code.mathparser.MathParser;
+import com.google.code.mathparser.MathParserFactory;
+import repository.Repository;
+import repository.RepositoryImpl;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static controllers.ControllerConstants.ABS_RANGE;
+import static controllers.ControllerConstants.DIVIDE;
+import static controllers.ControllerConstants.EMPTY_SYMBOL;
+import static controllers.ControllerConstants.EXPRESSION;
+import static controllers.ControllerConstants.MINUS;
+import static controllers.ControllerConstants.MULTIPLY;
+import static controllers.ControllerConstants.PLUS;
+import static controllers.ControllerConstants.STATUS_BAD_REQUEST;
+import static controllers.ControllerConstants.STATUS_CREATED;
+import static controllers.ControllerConstants.STATUS_FORBIDDEN;
+import static controllers.ControllerConstants.STATUS_OK;
+import static utils.ConversionUtil.deleteSpacesAndConvertListToString;
+
+
+public class CalculatorService implements Calculator {
+    private static final Repository REPOSITORY = RepositoryImpl.getInstance();
+
+    @Override
+    public int calculate(String ID) {
+        String expression = getFinalExpression(ID);
+        MathParser mathParser = MathParserFactory.create();
+
+        return mathParser.calculate(expression)
+                .doubleValue()
+                .intValue();
+    }
+
+    @Override
+    public int addData(String ID, String parameterName, String paramValue) {
+        String valueFromRepository = getValueFromRepository(ID, parameterName);
+        if (!isDataExists(ID)) {
+            REPOSITORY.putNewData(ID);
+        }
+        int statusCode = getStatusCode(valueFromRepository, parameterName, paramValue);
+        if (!isBadFormatExpression(paramValue, parameterName) && !isParameterHasOverLimitValue(paramValue)) {
+            REPOSITORY.update(ID, parameterName, paramValue);
+        }
+        return statusCode;
+    }
+
+    @Override
+    public void deleteData(String ID, String parameterName) {
+        Map<String, String> data = REPOSITORY.getDataByID(ID);
+
+        data.remove(parameterName);
+    }
+
+    private String getFinalExpression(String ID) {
+        List<String> expression = getExpression(ID);
+        Map<String, String> repositoryData = REPOSITORY.getDataByID(ID);
+
+        while (isExpressionWithVariables(repositoryData, expression)) {
+            convertExpressionWithValue(expression, ID);
+        }
+
+        return deleteSpacesAndConvertListToString(expression);
+    }
+
+    private List<String> getExpression(String ID) {
+        Map<String, String> data = REPOSITORY.getDataByID(ID);
+
+        return Arrays.asList(data
+                .get(EXPRESSION)
+                .split(EMPTY_SYMBOL));
+    }
+
+    private boolean isDataExists(String ID) {
+        return REPOSITORY.getRepositoryData().containsKey(ID);
```

Comment:
Maybe it would be better move this method to repository,
repository.existData(id)